.github/workflows/ci.yml

```diff
@@ -0,0 +1,172 @@
+name: CI
+on: [push, pull_request]
+defaults:
+  run:
+    working-directory: ./Software
+jobs:
+  build-ubuntu:
+    name: Ubuntu Build
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, ubuntu-20.04]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - run: sudo apt-get install qt5-default libqt5serialport5-dev build-essential libgtk2.0-dev libusb-1.0-0-dev libudev-dev libnotify-dev qttools5-dev-tools libappindicator-dev openssl libpulse-dev libfftw3-dev hashdeep
+      - run: qmake -r && make
+      - run: scripts/linux/prepare_installer.sh
+      - run: ./build-deb.sh amd64
+        working-directory: ./Software/dist_linux
+      - name: Check Package
+        run: |
+          set -xe
+          PKG_PATH=$(ls dist_linux/*.deb)
+          echo "PKG_PATH=Software/$PKG_PATH" >> $GITHUB_ENV
+          echo "PKG_NAME=$(basename $PKG_PATH | sed 's#amd64#${{ matrix.os }}_amd64#; s# ##')" >> $GITHUB_ENV
+      - name: Upload Package
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.PKG_NAME }}
+          path: ${{ env.PKG_PATH }}
+          
+  build-fw:
+    name: Firmware Build
+    runs-on: ubuntu-latest
+    steps:   
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+      - run: sudo apt-get install gcc-avr avr-libc build-essential
+      - run: ./build_batch.sh
+        working-directory: ./Firmware
+      - name: Archive artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: firmware
+          path: Firmware/hex/*
+          
+  build-win:
+    name: Windows Build
+    runs-on: windows-2019
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
```

Comment:
this thinks submodules are under the same user
`clone of 'https://github.com/zomfg/UpdateElevate.git' into submodule path 'D:/a/Lightpack/Lightpack/Software/UpdateElevate' failed`